packages/builder/src/actions.ts

```diff
@@ -46,29 +45,50 @@ export const ActionKinds: { [label: string]: CannonAction } = {};
 /**
  * NOTE: if you edit this schema, please also edit the constructor of `ChainDefinition` to account for non-action components of
  */
-const ChainDefinitionSchema = {
-  properties: {
-    name: { type: 'string' },
-    version: { type: 'string' },
-  },
-  optionalProperties: {
-    description: { type: 'string' },
-    keywords: { elements: { type: 'string' } },
-    setting: {
-      values: {
-        optionalProperties: {
-          description: { type: 'string' },
-          type: { enum: ['number', 'string', 'boolean'] },
-          defaultValue: { type: 'string' },
-        },
-      },
-    },
-
-    import: importSpec,
-  },
-} as const;
-
-export type RawChainDefinition = JTDDataType<typeof ChainDefinitionSchema>;
+
+const ChainDefinitionSchema = z
+  .object({
+    name: z.string({
+      required_error: 'name is required',
+      invalid_type_error: 'name must be a string',
+    }),
+    version: z.string({
+      required_error: 'version is required',
+      invalid_type_error: 'version must be a string',
+    }),
+  })
+  .merge(
+    z
+      .object({
+        description: z.string(),
+        keywords: z.array(
+          z.string({
+            invalid_type_error: 'keywords must be strings',
```

Comment:
Keywords must be an array of strings

Reply:
Yeah it is an array of strings, I'll change the error message to better resemble that but this error pops up if any of the keywords inside the array aren't strings.